3 - PB/Documentazione esterna/Analisi dei Requisiti/Analisi dei Requisiti.typ

```diff
@@ -343,920 +321,1408 @@ Questo documento è redatto in modo incrementale, così da risultare sempre conf
+)
+#printUseCaseInfo(
+  "Scenario principale",
+  "L'utente inserisce la categoria di prodotto desiderata"
+)
 
+=== Spostamento prodotto <ucd>
 #figure(
-  image("./imgs/uc11.png", format: "png", width: 60%),
-  caption: [
-    UML UC-11
-  ],
+  image("./imgs/UCd.svg", width: 120%),
+  caption: "Diagramma UC--d"
+)
+#printUseCaseInfo("Attore principale", "Utente")
+#printUseCaseInfo(
+  "Precondizioni",
+  "Il sistema è operativo",
+  "L'utente ha richiesto l'importazione dei prodotti",
+  "Il sistema ha importato correttamente almeno un prodotto"
+)
+#printUseCaseInfo(
+  "Postcondizioni",
+  "Il sistema inserisce l'ordine di movimentazione del prodotto verso il bin di destinazione"
+)
+#printUseCaseInfo(
+  "Scenario principale",
+  "L'utente richiede lo spostamento di un prodotto verso un bin di destinazione"
+)
+#printUseCaseInfo(
+  "User story associata",
+  "Come utente, desidero spostare un prodotto verso un bin di destinazione per poterlo (ri)collocare nell'ambiente 3D"
+)
+#printUseCaseInfo(
+  "Generalizzazioni",
+  [Spostamento prodotto non collocato in un bin (@ucd.1)],
+  [Spostamento prodotto collocato in un bin (@ucd.2)]
 )
 
-#derivedRequirements("UC-11")
-
-=== Spostamento della visuale lungo gli assi
-$bold("Descrizione: ")$
-successivamente alla configurazione dell'ambiente di lavoro (@uc1), l'utente può visualizzare il magazzino e spostare la visuale lungo almeno uno dei tre assi (orizzontale, verticale, longitudinale).
-
-$bold("Attore: ")$
-utente.
-
-$bold("Precondizioni: ")$
-- il sistema è stato correttamente configurato.
+==== Spostamento prodotto non collocato in un bin <ucd.1>
+#printUseCaseInfo(
+  "Generalizzazione di",
+  [Spostamento prodotto (@ucd)]
+)
+#printUseCaseInfo("Attore principale", "Utente")
+#printUseCaseInfo(
+  "Precondizioni",
+  "Il prodotto selezionato non è collocato in un bin",
+)
+#printUseCaseInfo(
+  "Inclusioni",
+  [Invio richiesta di movimentazione (@ucd.3)],
+)
 
-$bold("Postcondizioni: ")$
-- l'utente ha spostato la visuale sul magazzino nella direzione di almeno uno dei tre assi (orizzontale, verticale, longitudinale).
+==== Spostamento prodotto collocato in un bin <ucd.2>
+#printUseCaseInfo(
+  "Generalizzazione di",
+  [Spostamento prodotto (@ucd)]
+)
+#printUseCaseInfo("Attore principale", "Utente")
+#printUseCaseInfo(
+  "Precondizioni",
+  "Il prodotto selezionato è collocato in un bin",
+)
+#printUseCaseInfo(
+  "Estensioni",
+  [Invio richiesta di movimentazione (@ucd.3)],
+  [Visualizzazione errore spostamento impossibile (@ucd.2.1)]
+)
 
-$bold("Scenario: ")$
-- l'utente visualizza il magazzino;
-- l'utente può spostare la visuale del magazzino lungo l'asse verticale;
-- l'utente può spostare la visuale del magazzino lungo l'asse orizzontale;
-- l'utente può spostare la visuale del magazzino lungo l'asse longitudinale;
-- l'utente ha cambiato la prospettiva sul magazzino.
+===== Visualizzazione errore spostamento impossibile <ucd.2.1>
```

Comment:
Aggiungo quest'altra osservazione come commento: in questi UC la struttura della descrizione degli errori è diversa dagli UC visti ieri. Su quelli delle Zone la visualizzazione dell'errore è vista come postcondizione, mentre qui come scenario principale